cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/query/ForwardingDatastore.java

```diff
@@ -0,0 +1,181 @@
+package com.cloudant.sync.query;
+
+import com.cloudant.sync.datastore.Changes;
+import com.cloudant.sync.datastore.ConflictException;
+import com.cloudant.sync.datastore.ConflictResolver;
+import com.cloudant.sync.datastore.Datastore;
+import com.cloudant.sync.datastore.DocumentException;
+import com.cloudant.sync.datastore.DocumentNotFoundException;
+import com.cloudant.sync.datastore.DocumentRevision;
+import com.cloudant.sync.event.EventBus;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by rhys on 04/07/2016.
+ */
+public class ForwardingDatastore implements Datastore {
```

Comment:
I found the override usage in the MockMatcher now. A reflect proxy and invocation handler would still be a lot less code.